Fix diff opts handling (ignore case / whitespace)

wwade opened the pull request:
Inside `vdiff-refresh`, when append is used to build up the diff command, if one of the intermediate args to append is a string and not a list, append will expand it into the string's character values.

So, "-w" becomes 45 119, and this `Wrong type argument: stringp, 45` error message is shown.

Added a small helper that converts non-empty strings using `ensure-list` to avoid this problem. We only want non-empty strings, though, otherwise an empty diff opt would get added as an explicit empty string arg which would cause diff to fail, e.g.

    diff -w "" -- file1 file2

Where we actually want:

    diff -w -- file1 file2

Fixes: #36

wwade commented:
@justbur when you have a moment :smile: 

I ran the tests on my fork and they pased FYI.

justbur commented:
Thanks for this. It seems like it would be more consistent/logical to make `vdiff-session-whitespace-args` a list everywhere. What do you think?

wwade commented:
Sounds reasonable, let me a have a crack at that.

wwade committed:
make vdiff-session-whitespace-args a list everywhere

@justbur suggested making vdiff-session-whitespace-args a list
everywhere instead of converting it to a list when it is being used.

I also changed vdiff-session-case-args in the same way.

wwade committed:
Fix diff opts handling (ignore case / whitespace)

Inside `vdiff-refresh`, when append is used to build up the diff
command, if one of the intermediate args to append is a string and not
a list, append will expand it into the string's character values.

So, "-w" becomes 45 119, and this `Wrong type argument: stringp, 45`
error message is shown.

Added a small helper that converts non-empty strings using
`ensure-list` to avoid this problem. We only want non-empty strings,
though, otherwise an empty diff opt would get added as an explicit
empty string arg which would cause diff to fail, e.g.

    diff -w "" -- file1 file2

Where we actually want:

    diff -w -- file1 file2

Fixes: #36

wwade committed:
fix docstring for vdiff--maybe-list

wwade commented:
> Thanks for this. It seems like it would be more consistent/logical to make `vdiff-session-whitespace-args` a list everywhere. What do you think?

OK, it's finished at last!

justbur commented:
thanks
